# Description: Fix build on non-Android ARM Linux builds without NEON
# Author: Chris Coulson <chris.coulson@canonical.com>

diff --git a/third_party/webrtc/build/common.gypi b/third_party/webrtc/build/common.gypi
--- a/third_party/webrtc/build/common.gypi
+++ b/third_party/webrtc/build/common.gypi
@@ -234,17 +234,21 @@
         ],
         'conditions': [
           ['arm_version==7', {
             'defines': ['WEBRTC_ARCH_ARM_V7',],
             'conditions': [
               ['arm_neon==1', {
                 'defines': ['WEBRTC_ARCH_ARM_NEON',],
               }, {
-                'defines': ['WEBRTC_DETECT_ARM_NEON',],
+                'conditions': [
+                  ['arm_neon_optional==1', {
+                    'defines': ['WEBRTC_DETECT_ARM_NEON',],
+                  }],
+                ],
               }],
             ],
           }],
         ],
       }],
       ['target_arch=="mipsel"', {
         'defines': [
           'MIPS32_LE',
diff --git a/third_party/webrtc/common_audio/resampler/sinc_resampler.cc b/third_party/webrtc/common_audio/resampler/sinc_resampler.cc
--- a/third_party/webrtc/common_audio/resampler/sinc_resampler.cc
+++ b/third_party/webrtc/common_audio/resampler/sinc_resampler.cc
@@ -128,25 +128,29 @@ void SincResampler::InitializeCPUSpecifi
 void SincResampler::InitializeCPUSpecificFeatures() {
   convolve_proc_ = WebRtc_GetCPUInfo(kSSE2) ? Convolve_SSE : Convolve_C;
 }
 #endif
 #elif defined(WEBRTC_ARCH_ARM_V7)
 #if defined(WEBRTC_ARCH_ARM_NEON)
 #define CONVOLVE_FUNC Convolve_NEON
 void SincResampler::InitializeCPUSpecificFeatures() {}
-#else
+#elif defined(WEBRTC_DETECT_ARM_NEON)
 // ARM CPU detection required.  Function will be set by
 // InitializeCPUSpecificFeatures().
 #define CONVOLVE_FUNC convolve_proc_
 
 void SincResampler::InitializeCPUSpecificFeatures() {
   convolve_proc_ = WebRtc_GetCPUFeaturesARM() & kCPUFeatureNEON ?
       Convolve_NEON : Convolve_C;
 }
+#else
+// No NEON
+#define CONVOLVE_FUNC Convolve_C
+void SincResampler::InitializeCPUSpecificFeatures() {}
 #endif
 #else
 // Unknown architecture.
 #define CONVOLVE_FUNC Convolve_C
 void SincResampler::InitializeCPUSpecificFeatures() {}
 #endif
 
 SincResampler::SincResampler(double io_sample_rate_ratio,
